### PR TITLE
Refactorings minor fixes

### PR DESCRIPTION
**Pull request by MadhavVij**
- Better error handling
- using Defer and closing connection calls early in the stack
- Added context manager and improved log messages
- Minor fixes and f-strings implementations

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -64,12 +65,10 @@ func Router(_ context.Context, event events.APIGatewayV2HTTPRequest) (*events.AP
...
 // routeSubmitFeedback - Receives feedback from Stellar clients, and stores the entries in a DynamoDB table
 func routeSubmitFeedback(event events.APIGatewayV2HTTPRequest) (*events.APIGatewayProxyResponse, error) {
-	b := []byte(event.Body)
 	var clientFeedback ClientFeedback
-	err := json.Unmarshal(b, &clientFeedback)
-
+	err := json.Unmarshal([]byte(event.Body), &clientFeedback)
 	if err != nil {
-		return handleInternalError("Could not parse post body: %s", err)
+		return handleInternalError(fmt.Sprintf("Could not parse post body: %s", err), nil)
```
Comment (StellarRedditApp) [inline main.go:70]: Good logical catch. We decided not to pass the error to the client, so the field descriptor `%s` should not be there.

**Review by StellarRedditApp — COMMENTED** [review]
image

File: main.go [hunk 2]
```diff
@@ -226,11 +225,10 @@ func routeDeleteFeedback(event events.APIGatewayV2HTTPRequest) (*events.APIGatew
...
 	formattedError := fmt.Sprintf("%s - %s", detail, err)
-	finalError := errors.New(formattedError)
 	fmt.Println(formattedError)
 
 	return &events.APIGatewayProxyResponse{
 		StatusCode: http.StatusInternalServerError,
-		Body:       detail,
-	}, finalError
+		Body:       formattedError,
+	}, fmt.Errorf("%s - %w", detail, err)
```
Comment (StellarRedditApp) [inline main.go:233]: Specialized formatter! Looks good.

**Comment by aciodev** [comment]
Several functions were called that do not exist. Certain changes have been reverted as a result. See additional commits prior to merge.

**Pull request merged**